pkg/liqo-controller-manager/external-network/wireguard/wggatewayserver_controller.go

```diff
@@ -384,6 +385,20 @@ func (r *WgGatewayServerReconciler) forgeEndpointStatusNodePort(ctx context.Cont
 		}
 	}
 
+	if service.Annotations != nil {
+		if v, ok := service.Annotations[consts.OverrideAddressAnnotation]; ok {
+			addresses = []string{v}
+		}
+		if v, ok := service.Annotations[consts.OverridePortAnnotation]; ok {
+			p, err := strconv.ParseInt(v, 10, 32)
+			if err != nil {
+				klog.Errorf("unable to parse port %q from service %s/%s annotation: %v", v, service.Namespace, service.Name, err)
+				return nil, nil, err
+			}
+			port = int32(p)
+		}
+	}
```

Comment:
What about creating a function in utils.go to avoid duplicate code? (i.e., L452-L464)